### ISSUE TITLE
Git(Hub|Lab) APIs and GNOME

### ISSUE THREAD

**Issue by yarons** [release-monitoring/anitya#567]
I looked at the current plugin and maybe fetching the entire HTML is a little counter intuitive.

With the latest GitHub GraphQL I can fetch the latest tag like this:

```
query {
  repository(owner: "release-monitoring", name: "anitya") {
    tags: refs(refPrefix: "refs/tags/", first: 1, direction: DESC) {
      edges {
        tag: node {
          name
        }
      }
    }
  }
}
```
Which is very efficient but requires a user.
With v3 it's possible to fetch all tags like this:
https://api.github.com/repos/release-monitoring/anitya/tags

It works pretty similar with GitLab:
https://gitlab.gnome.org/api/v4/projects/GNOME%2Fgnome-video-arcade/repository/tags

GNOME recently switch to GitLab to it's better to remove that in favor of a native GitLab plugin as well (Or just redirect it properly).

**Comment by jeremycline** [comment]
Indeed, and it'd be much better to use APIs. https://github.com/release-monitoring/anitya/pull/502 started on an implementation using the v3 (I think) API. I think someone could pick that PR up and make a few changes and it'd be good to go. Same deal with GitLab.

**Comment by Zlopez** [comment]
I will start on implementation of GitHub API [v4](https://developer.github.com/v4/) instead of using the older v3.

I see that the GitLab API is now only supporting REST and the GitHub API is GraphQL. So this looks like this will be two different APIs to implement.

**Comment by Zlopez** [comment]
Github and Gitlab API were already merged. Closing this issue.

**Comment by yarons** [comment]
So now we'll just have to wait for the production version upgrade?

**Comment by Zlopez** [comment]
I need to release new version first and deploy it in staging.

But yes, we need to wait till upgrade on production.

**Comment by yarons** [comment]
Is it in 0.12.1?

**Comment by Zlopez** [comment]
Just now only in master, it will be released in 0.13.0.

**Comment by yarons** [comment]
Regardless, thank you so much for this, keep up the great work!
